main.go

```diff
@@ -95,13 +136,41 @@ func (g Gaz) Run() {
 
 		// register /version to return the build version
 		g.Router.HandleFunc("/version", VersionHTML).Methods("GET")
-
-		port := viper.GetInt("http.port")
-		Sugar.Infof("Starting http server on port %d", port)
-		err := http.ListenAndServe(fmt.Sprintf(":%d", port), g.Router)
+		Sugar.Infof("Starting HTTP server on :%d", g.httpPort)
+		err := http.Serve(*g.httpListener, g.Router)
 		if err != nil {
-			Sugar.Errorf("Cannot start HTTP server on :%d, %+v", port, err)
+			Sugar.Errorf("Cannot updater HTTP server on :%d, %+v", g.httpPort, err)
 			panic(err)
 		}
 	}()
 }
+
+func (g Gaz) Register(d *ServiceDefinition) (string, error) {
+	if g.ServiceDiscovery == nil {
+		return "", errors.New("no service registry configured")
+	}
+	return g.ServiceDiscovery.Register(d, g.httpPort)
```

Comment:
httpPort? not the gRPC port instead?

Reply:
EDIT:
I start to understand as I go down my review. For the consul health check, we need the http port. Am I correct?

Reply:
yes, but I might replace it with the gRPC health check